mitodl/mit-open
# Shanbady/topic channel page header fixes

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4542 and https://github.com/mitodl/hq/issues/4540 and https://github.com/mitodl/mit-open/issues/1045



### Description (What does it do?)
This PR fixes 3 sets of issues. Some style changes to the channel page banner make it more inline with the figma (see https://github.com/mitodl/hq/issues/4542 and https://github.com/mitodl/hq/issues/4540) and the addition of a configurable dark overlay to channel page banners (https://github.com/mitodl/mit-open/issues/1045)

### Screenshots (if appropriate):

<img width="1386" alt="Screenshot 2024-06-12 at 9 43 40 AM" src="https://github.com/mitodl/mit-open/assets/196425/46ca16db-5cf7-45f0-a28e-81c0674837f8">
<img width="1387" alt="Screenshot 2024-06-12 at 9 43 55 AM" src="https://github.com/mitodl/mit-open/assets/196425/58b71aa9-f968-4439-b706-f056c2822140">
<img width="1677" alt="Screenshot 2024-06-12 at 9 48 23 AM" src="https://github.com/mitodl/mit-open/assets/196425/c08f64c3-ade8-4c2b-ac25-7147e4e6571f">

<img width="517" alt="Screenshot 2024-06-12 at 9 46 13 AM" src="https://github.com/mitodl/mit-open/assets/196425/2e7ce4b4-fddb-4f53-a8f7-d249e9c3a21d">
<img width="520" alt="Screenshot 2024-06-12 at 9 50 47 AM" src="https://github.com/mitodl/mit-open/assets/196425/2ef74858-982e-4b44-9fe7-7c48be675dce">
<img width="519" alt="Screenshot 2024-06-12 at 9 51 27 AM" src="https://github.com/mitodl/mit-open/assets/196425/e6acedd4-d68b-49a1-8203-bf10dd8b3ba9">


### How can this be tested?
1. checkout this branch and make sure units/offerors and topics are loaded 
  ```
   python manage.py update_offered_by
   python manage.py backpopulate_resource_channels --overwrite --unit
   python manage.py update_platforms
   python manage.py update_topics
   ```
3. login and visit the various channel pages like http://od.odl.local:8063/c/unit/see http://od.odl.local:8063/c/unit/xpro http://od.odl.local:8063/c/unit/bootcamps http://od.odl.local:8063/c/department/history and http://od.odl.local:8063/c/topic/physics
4. validate that the pages look like the figma mocks: 

desktop:
  - https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=3774-68962&viewport=21%2C-3032%2C0.89&t=Hy7uaWCxhWB7TXwN-1&scaling=min-zoom&starting-point-node-id=2197%3A31345
  - https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=2197-31446&viewport=21%2C-3032%2C0.89&t=Hy7uaWCxhWB7TXwN-1&scaling=min-zoom&starting-point-node-id=2197%3A31446
   

mobile:
  - https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=4083-96092&viewport=21%2C-3032%2C0.89&t=Hy7uaWCxhWB7TXwN-1&scaling=min-zoom&starting-point-node-id=2197%3A31345
  - https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=2197-31345&viewport=21%2C-3032%2C0.89&t=Hy7uaWCxhWB7TXwN-1&scaling=min-zoom&starting-point-node-id=2197%3A31345&show-proto-sidebar=1
